src/main/java/org/sunbird/org/service/ExtendedOrgServiceImpl.java

```diff
@@ -1079,4 +1081,51 @@ private void fetchStateOrMinistryDetails(Map<String, Object> requestBody) {
 			}
 		}
 	}
+
+	@Override
+	public SBApiResponse updateV2(Map<String, Object> orgRequest, String userToken) {
+		logger.info("ExtendedOrgServiceImpl::updateV2::Starting the update of the organization");
+		SBApiResponse outgoingResponse = ProjectUtil.createDefaultResponse(Constants.API_ORG_EXT_UPDATE);
+		String errMsg = validateRequestFieldsV2(orgRequest, outgoingResponse);
+		if (!StringUtils.isEmpty(errMsg)) return outgoingResponse;
+		OrgHierarchyInfo orgHierarchyInfo = new OrgHierarchyInfo();
+		if(!StringUtils.isBlank(MapUtils.getString(orgRequest, Constants.ORG_NAME))){
+			orgRepository.updateOrgNameBySbOrgId((String) orgRequest.get(Constants.ORG_ID), (String) orgRequest.get(Constants.ORG_NAME));
+			orgHierarchyInfo.setOrgName((String) orgRequest.get(Constants.ORG_NAME));
+		}
+		orgHierarchyInfo.setSbOrgId((String) orgRequest.get(Constants.ORG_ID));
+		Map<String, Object> orgDataUpdateResonse = updateOrgDetailsToDB(userToken, orgHierarchyInfo, orgRequest);
+		if (MapUtils.isEmpty(orgDataUpdateResonse) || !orgDataUpdateResonse.get(Constants.RESPONSE_CODE).equals(Constants.OK)) {
+			logger.info("ExtendedOrgServiceImpl::updateV2::Failed to update Org details for organization: " + orgHierarchyInfo.getSbOrgId());
+			setInternalServerError(outgoingResponse, "Error while updating the organization details");
+		} else {
+			populateSuccessResponse(outgoingResponse);
+		}
+		return outgoingResponse;
+	}
+
+	private String validateRequestFieldsV2(Map<String, Object> request, SBApiResponse response) {
+		if (StringUtils.isBlank(MapUtils.getString(request, Constants.ORG_ID))) {
+			response.getParams().setStatus(Constants.FAILED);
+			response.getParams().setErrmsg("Organization ID is missing");
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			return "Organization ID is missing";
+		}
+		for (String fieldName : request.keySet()) {
+			if (!validateUpdatableFields(fieldName)) {
+				logger.info("ExtendedOrgServiceImpl::updateV2::Invalid field in request: {}", fieldName);
+				response.getParams().setStatus("Field : " + fieldName + " is not allowed to be updated.");
```

Comment:
Status should say -- Success / Failed
Do add errMsg and make sure that contains proper details